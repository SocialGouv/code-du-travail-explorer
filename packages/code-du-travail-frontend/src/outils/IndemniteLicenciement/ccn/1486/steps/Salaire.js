import React from "react";
import { Field } from "react-final-form";
import { FieldArray } from "react-final-form-arrays";
import styled from "styled-components";
import { subMonths, format } from "date-fns";
import frLocale from "date-fns/locale/fr";
import { Table as UITable, Toast, theme } from "@socialgouv/react-ui";

import { parse } from "../../../../common/date";
import { Input } from "../../../../common/stepStyles";
import { InlineError } from "../../../../common/ErrorField";
import { isNumber } from "../../../../common/validators";
import { YesNoQuestion } from "../../../../common/YesNoQuestion";

export const Salaire = ({ form }) => {
  const data = form.getState().values;
  return (
    <>
      <Notice />
      <YesNoQuestion
        name="hasBrancheNewSalaire"
        label="De ce fait, avez-vous une modification à faire dans la déclaration du salaire&nbsp;?"
      />
      {data.hasBrancheNewSalaire && (
        <>
          <YesNoQuestion
            name="hasBrancheNewRegularSalaire"
            label="Le salaire mensuel a-t-il été le même durant les 12 derniers mois&nbsp;?"
            onChange={hasSameSalaire => {
              if (hasSameSalaire) {
                form.change("brancheNewIrregularSalaire", undefined);
              } else {
                form.change(
                  "brancheNewIrregularSalaire",
                  Array.from({ length: 12 }).map((_, index) => ({
                    label: format(
                      subMonths(parse(data.dateSortie), index),
                      "MMMM yyyy",
                      {
                        locale: frLocale
                      }
                    ),
                    salary: null
                  }))
                );
                form.change("brancheNewRegularSalaire", undefined);
              }
            }}
          />
          {data.hasBrancheNewRegularSalaire === true && (
            <RegularNewSalaire name="brancheNewRegularSalaire" />
          )}

          <IrregularNewSalaire name="brancheNewIrregularSalaire" />
        </>
      )}
    </>
  );
};

const RegularNewSalaire = ({ name }) => (
  <>
    <Field
      name={name}
      validate={isNumber}
      subscription={{
        value: true,
        error: true,
        touched: true,
        invalid: true
      }}
    >
      {({ input, meta: { touched, error, invalid } }) => (
        <>
          <p>Salaire mensuel ajusté (primes et temps partiel inclus)</p>
          <CurrencyWrapper>
            <NumberInput
              {...input}
              size="10"
              type="number"
              invalid={touched && invalid}
            />
            <Currency aria-hidden="true">€</Currency>
          </CurrencyWrapper>
          {error && touched && invalid && <InlineError>{error}</InlineError>}
        </>
      )}
    </Field>
  </>
);

const IrregularNewSalaire = ({ name }) => (
  <>
    <FieldArray name={name}>
      {({ fields }) =>
        fields.length > 0 && (
          <Table>
            <thead>
              <tr>
                <th>Mois</th>
                <th>Salaire mensuel ajusté (primes et temps partiel inclus)</th>
              </tr>
            </thead>
            <tbody>
              {fields.map((name, index) => (
                <tr key={name}>
                  <td>
                    <label htmlFor={`salaire${index}`}>
                      {fields.value[index].label}
                    </label>
                  </td>
                  <td>
                    <Field
                      name={`${name}.salary`}
                      validate={isNumber}
                      formatOnBlur
                      format={value => {
                        // Hack auto fill
                        fields.value.forEach((field, fieldIndex) => {
                          if (
                            fieldIndex > index &&
                            fields.value[fieldIndex].salary === null
                          ) {
                            fields.value[fieldIndex].salary = isFinite(value)
                              ? parseFloat(value)
                              : null;
                          }
                        });
                        return isFinite(value) ? parseFloat(value) : null;
                      }}
                      subscription={{
                        value: true,
                        error: true,
                        touched: true,
                        invalid: true
                      }}
                      render={({
                        input,
                        meta: { touched, error, invalid }
                      }) => (
                        <>
                          <CurrencyWrapper>
                            <NumberInput
                              {...input}
                              size="10"
                              type="number"
                              id={`salaire${index}`}
                              invalid={touched && invalid}
                            />
                            <Currency aria-hidden="true">€</Currency>
                          </CurrencyWrapper>
                          {error && touched && invalid && (
                            <InlineError>{error}</InlineError>
                          )}
                        </>
                      )}
                    />
                  </td>
                </tr>
              ))}
            </tbody>
          </Table>
        )
      }
    </FieldArray>
  </>
);

const Notice = () => (
  <>
    <p>Le salaire à renseigner pour cette convention collective inclut:</p>
    <ul>
      <li>les primes prévues au contrat de travail</li>
    </ul>
    <strong>mais elle exclut</strong>:
    <ul>
      <li>
        les heures supplémentaires au delà de l’heure normale de l’entreprise
      </li>
      <li>
        les majorations/indemnités liées à un déplacement ou un détachement
      </li>
    </ul>
    <StyledToast variant="warning">
      Attention, les éventuelles <strong>primes</strong> renseignées
      précédemment ne seront pas prises en compte dans le calcul de l’indemnité
      de la convention collective. Si des primes ont été perçues, merci de les
      intégrer à la rémunération ci-après.
      <br />
      Aussi, si il y en a, les mois effectués en <strong>
        temps partiel
      </strong>{" "}
      sont à considérer comme ceux effectués en temps plein.
    </StyledToast>
  </>
);

const { colors, spacing } = theme;

const StyledToast = styled(Toast)`
  margin-top: ${spacing.interComponent};
`;

const Table = styled(UITable)`
  & tr > td:nth-child(2) {
    width: 70%;
    text-align: left;
  }
`;
const NumberInput = styled(Input)`
  padding-right: ${spacing.base};
  text-align: right;
`;

const CurrencyWrapper = styled.div`
  position: relative;
  display: inline-block;
  margin-right: ${spacing.interComponent};
`;

const Currency = styled.span`
  position: absolute;
  top: 50%;
  right: 0.25rem;
  color: ${colors.grey};
  transform: translateY(-50%);
`;
