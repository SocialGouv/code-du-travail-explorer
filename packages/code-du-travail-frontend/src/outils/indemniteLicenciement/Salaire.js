import React from "react";
import PropTypes from "prop-types";
import { Container, Section } from "@cdt/ui";

import { PrevNextStepper } from "./PrevNextStepper";
import { inputStyle } from "./stepStyles";

const times = num => Array.from({ length: num }, (_, i) => i);

class Salaire extends React.Component {
  static propTypes = {
    onChange: PropTypes.func,
    value: PropTypes.arrayOf(PropTypes.number).isRequired,
    onPrevious: PropTypes.func.isRequired,
    onNext: PropTypes.func.isRequired,
    nextDisabled: PropTypes.bool
  };

  static defaultProps = {
    value: Array.from({ length: 12 }, () => 0),
    nextDisabled: false
  };

  constructor(props, ...args) {
    super(props, ...args);
    this.state = {
      salaires: props.value
    };
  }

  onSalaireClick = i => e => {
    const value = parseFloat(e.target.value);
    if (value === 0) {
      this.updateSalaireValue(i, "");
    }
  };

  onSalaireBlur = i => e => {
    // update below fields values when empty
    const value = parseFloat(e.target.value);
    if (!value) {
      this.updateSalaireValue(i, 0);
      return;
    }
    const curSalaires = [...this.state.salaires];
    let j = i + 1;
    while (j < curSalaires.length) {
      if (parseFloat(curSalaires[j]) > 0) {
        break;
      }
      curSalaires[j] = value;
      j++;
    }
    this.setState(
      {
        salaires: curSalaires
      },
      () => {
        if (this.props.onChange) {
          this.props.onChange(this.state.salaires);
        }
      }
    );
  };
  updateSalaireValue = (index, value, cb) => {
    this.setState(curState => {
      const curSalaires = [...curState.salaires];
      curSalaires[index] = value;
      return {
        salaires: curSalaires
      };
    }, cb);
  };
  updateSalaire = i => e => {
    // update a single row
    const value = parseFloat(e.target.value) || 0;
    this.updateSalaireValue(i, value, () => {
      if (this.props.onChange) {
        this.props.onChange(this.state.salaires);
      }
    });
  };
  render() {
    const { onPrevious, onNext, nextDisabled } = this.props;
    return (
      <React.Fragment>
        <Section light>
          <React.Fragment>
            <h2>Derniers mois de salaire</h2>
            <table width="100%" style={{ fontSize: "1.2em" }}>
              <thead>
                <tr>
                  <td
                    width={140}
                    style={{ textAlign: "center", fontWeight: "bold" }}
                  >
                    Mois
                  </td>
                  <td
                    width={200}
                    style={{ textAlign: "center", fontWeight: "bold" }}
                  >
                    Salaire brut mensuel
                  </td>
                </tr>
              </thead>
              <tbody>
                {times(12).map(i => (
                  <tr key={i} style={{ background: i < 3 ? "#ddd" : "" }}>
                    <td>{i === 0 ? "Dernier salaire" : `Salaire N-${i}`}</td>
                    <td style={{ textAlign: "center" }}>
                      <input
                        onBlur={this.onSalaireBlur(i)}
                        onClick={this.onSalaireClick(i)}
                        value={this.state.salaires[i]}
                        onChange={this.updateSalaire(i)}
                        style={{ ...inputStyle, width: 100 }}
                        type="number"
                      />
                    </td>
                    {i === 0 && (
                      <td
                        valign="middle"
                        style={{ textAlign: "center", background: "#ddd" }}
                        rowSpan={3}
                      >
                        trois derniers
                        <br />
                        mois de salaire
                      </td>
                    )}
                  </tr>
                ))}
              </tbody>
            </table>
          </React.Fragment>
        </Section>
        <Container>
          <PrevNextStepper
            onPrev={onPrevious}
            onNext={onNext}
            nextDisabled={nextDisabled}
          />
        </Container>
      </React.Fragment>
    );
  }
}

export { Salaire };
