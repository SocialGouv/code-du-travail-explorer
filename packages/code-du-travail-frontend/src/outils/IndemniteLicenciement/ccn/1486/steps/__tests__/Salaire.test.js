import { render } from "@testing-library/react";
import arrayMutators from "final-form-arrays";
import React from "react";
import { Form } from "react-final-form";

import { Salaire } from "../Salaire";

describe("<Salaire />", () => {
  it("renders", () => {
    const onSubmit = jest.fn();
    const { container } = render(
      <Form
        onSubmit={onSubmit}
        mutators={{
          ...arrayMutators,
        }}
      >
        {({ form }) => <Salaire form={form} />}
      </Form>
    );
    expect(container).toMatchSnapshot();
  });
  it("renders a question asking about income's regularity", () => {
    const onSubmit = jest.fn();
    const initialValues = { hasBrancheNewSalaire: true };
    const { container } = render(
      <Form
        onSubmit={onSubmit}
        mutators={{
          ...arrayMutators,
        }}
        initialValues={initialValues}
      >
        {({ form }) => <Salaire form={form} />}
      </Form>
    );
    expect(container).toMatchSnapshot();
  });
  it("renders an input field of type number", () => {
    const onSubmit = jest.fn();
    const initialValues = {
      hasBrancheNewRegularSalaire: true,
      hasBrancheNewSalaire: true,
    };
    const { container } = render(
      <Form
        onSubmit={onSubmit}
        mutators={{
          ...arrayMutators,
        }}
        initialValues={initialValues}
      >
        {({ form }) => <Salaire form={form} />}
      </Form>
    );
    expect(container).toMatchSnapshot();
  });
  it("renders a table of fields of type number", () => {
    const onSubmit = jest.fn();
    const initialValues = {
      brancheNewIrregularSalaire: [
        {
          label: "juillet 2019",
          salary: null,
        },
        {
          label: "juin 2019",
          salary: null,
        },
      ],
      hasBrancheNewRegularSalaire: false,
      hasBrancheNewSalaire: true,
    };
    const { container } = render(
      <Form
        onSubmit={onSubmit}
        mutators={{
          ...arrayMutators,
        }}
        initialValues={initialValues}
      >
        {({ form }) => <Salaire form={form} />}
      </Form>
    );
    expect(container).toMatchSnapshot();
  });
});
