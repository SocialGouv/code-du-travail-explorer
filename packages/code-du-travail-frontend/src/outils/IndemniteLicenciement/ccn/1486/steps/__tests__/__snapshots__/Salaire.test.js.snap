// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Salaire /> renders 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: auto;
  min-height: 48px;
  color: #3e486e;
  background-color: #fff;
  border-color: #7994d4;
  border-style: solid;
  border-width: 2px;
  border-radius: 0.6rem;
  box-shadow: none;
  -webkit-animation: none;
  animation: none;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 1rem 1.6rem;
  text-align: left;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2em;
  padding: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 2rem;
}

.c3 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: default;
}

.c4 {
  display: inline-block;
  margin-left: 1rem;
  color: #eb5757;
}

.c1 {
  margin-top: 2rem;
}

@media (max-width:600px) {
  .c3 {
    font-size: 1.6rem;
  }
}

<div>
  <p>
    Le salaire à renseigner pour cette convention collective inclut:
  </p>
  <ul>
    <li>
      les primes prévues au contrat de travail
    </li>
  </ul>
  <strong>
    mais elle exclut
  </strong>
  :
  <ul>
    <li>
      les heures supplémentaires au delà de l’heure normale de l’entreprise
    </li>
    <li>
      les majorations/indemnités liées à un déplacement ou un détachement
    </li>
  </ul>
  <div
    class="c0 c1"
  >
    <div
      class="c2"
      role="alert"
    >
      Attention, les éventuelles 
      <strong>
        primes
      </strong>
       renseignées précédemment ne seront pas prises en compte dans le calcul de l’indemnité de la convention collective. Si des primes ont été perçues, merci de les intégrer à la rémunération ci-après.
      <br />
      Aussi, si il y en a, les mois effectués en 
      <strong>
        temps partiel
      </strong>
       
      sont à considérer comme ceux effectués en temps plein.
    </div>
  </div>
  <p
    class="c3"
  >
    De ce fait, avez-vous une modification à faire dans la déclaration du salaire ?
    <span
      aria-label="champs obligatoire"
      class="c4"
    >
      *
    </span>
  </p>
  <div
    class="c5"
  >
    <label
      class="c6"
    >
      <input
        name="hasBrancheNewSalaire"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c6"
    >
      <input
        name="hasBrancheNewSalaire"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
</div>
`;

exports[`<Salaire /> renders a question asking about income's regularity 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: auto;
  min-height: 48px;
  color: #3e486e;
  background-color: #fff;
  border-color: #7994d4;
  border-style: solid;
  border-width: 2px;
  border-radius: 0.6rem;
  box-shadow: none;
  -webkit-animation: none;
  animation: none;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 1rem 1.6rem;
  text-align: left;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2em;
  padding: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 2rem;
}

.c3 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: default;
}

.c4 {
  display: inline-block;
  margin-left: 1rem;
  color: #eb5757;
}

.c1 {
  margin-top: 2rem;
}

@media (max-width:600px) {
  .c3 {
    font-size: 1.6rem;
  }
}

<div>
  <p>
    Le salaire à renseigner pour cette convention collective inclut:
  </p>
  <ul>
    <li>
      les primes prévues au contrat de travail
    </li>
  </ul>
  <strong>
    mais elle exclut
  </strong>
  :
  <ul>
    <li>
      les heures supplémentaires au delà de l’heure normale de l’entreprise
    </li>
    <li>
      les majorations/indemnités liées à un déplacement ou un détachement
    </li>
  </ul>
  <div
    class="c0 c1"
  >
    <div
      class="c2"
      role="alert"
    >
      Attention, les éventuelles 
      <strong>
        primes
      </strong>
       renseignées précédemment ne seront pas prises en compte dans le calcul de l’indemnité de la convention collective. Si des primes ont été perçues, merci de les intégrer à la rémunération ci-après.
      <br />
      Aussi, si il y en a, les mois effectués en 
      <strong>
        temps partiel
      </strong>
       
      sont à considérer comme ceux effectués en temps plein.
    </div>
  </div>
  <p
    class="c3"
  >
    De ce fait, avez-vous une modification à faire dans la déclaration du salaire ?
    <span
      aria-label="champs obligatoire"
      class="c4"
    >
      *
    </span>
  </p>
  <div
    class="c5"
  >
    <label
      class="c6"
    >
      <input
        checked=""
        name="hasBrancheNewSalaire"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c6"
    >
      <input
        name="hasBrancheNewSalaire"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
  <p
    class="c3"
  >
    Le salaire mensuel a-t-il été le même durant les 12 derniers mois ?
    <span
      aria-label="champs obligatoire"
      class="c4"
    >
      *
    </span>
  </p>
  <div
    class="c5"
  >
    <label
      class="c6"
    >
      <input
        name="hasBrancheNewRegularSalaire"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c6"
    >
      <input
        name="hasBrancheNewRegularSalaire"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
</div>
`;

exports[`<Salaire /> renders a table of fields of type number 1`] = `
.c7 {
  position: relative;
  overflow: hidden;
}

.c7:before,
.c7:after {
  position: absolute;
  top: 0;
  bottom: 0;
  display: block;
  width: 4rem;
  opacity: 0;
  -webkit-transition: opacity 0.3s linear;
  transition: opacity 0.3s linear;
  content: "";
  pointer-events: none;
  background: radial-gradient( ellipse at center, #fff 15%, rgba(255,255,255,0) 80% );
}

.c7:before {
  left: -2rem;
}

.c7:after {
  right: -2rem;
}

.c8 {
  overflow-x: auto;
}

.c9 {
  text-align: left;
  empty-cells: show;
  background-color: #fff;
  border: 1px solid #bbcadf;
  border-collapse: collapse;
  border-spacing: 0;
}

.c9 caption {
  padding: 1rem 0;
  font-style: italic;
  text-align: center;
}

.c9 td,
.c9 th {
  padding: 1rem 1.6rem;
  border: 1px solid #bbcadf;
}

.c9 th {
  color: #2f3b6c;
  font-weight: 600;
  background: #e4e8ef;
}

.c9 thead,
.c9 tfoot {
  padding: 1rem 1.6rem;
  vertical-align: bottom;
  background: #f2f5fa;
}

.c9 tfoot {
  vertical-align: top;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: auto;
  min-height: 48px;
  color: #3e486e;
  background-color: #fff;
  border-color: #7994d4;
  border-style: solid;
  border-width: 2px;
  border-radius: 0.6rem;
  box-shadow: none;
  -webkit-animation: none;
  animation: none;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 1rem 1.6rem;
  text-align: left;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2em;
  padding: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 2rem;
}

.c3 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: default;
}

.c4 {
  display: inline-block;
  margin-left: 1rem;
  color: #eb5757;
}

.c1 {
  margin-top: 2rem;
}

.c10 tr > td:nth-child(2) {
  width: 70%;
  text-align: left;
}

.c12 {
  width: 10em;
  text-align: right;
  border-color: #bbcadf;
  padding-right: 1.6rem;
  text-align: right;
}

.c12::-webkit-outer-spin-button,
.c12::-webkit-inner-spin-button {
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c11 {
  position: relative;
  display: inline-block;
  margin-right: 2rem;
}

.c13 {
  position: absolute;
  top: 50%;
  right: 0.25rem;
  color: #4d73b8;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

@media (max-width:600px) {
  .c3 {
    font-size: 1.6rem;
  }
}

<div>
  <p>
    Le salaire à renseigner pour cette convention collective inclut:
  </p>
  <ul>
    <li>
      les primes prévues au contrat de travail
    </li>
  </ul>
  <strong>
    mais elle exclut
  </strong>
  :
  <ul>
    <li>
      les heures supplémentaires au delà de l’heure normale de l’entreprise
    </li>
    <li>
      les majorations/indemnités liées à un déplacement ou un détachement
    </li>
  </ul>
  <div
    class="c0 c1"
  >
    <div
      class="c2"
      role="alert"
    >
      Attention, les éventuelles 
      <strong>
        primes
      </strong>
       renseignées précédemment ne seront pas prises en compte dans le calcul de l’indemnité de la convention collective. Si des primes ont été perçues, merci de les intégrer à la rémunération ci-après.
      <br />
      Aussi, si il y en a, les mois effectués en 
      <strong>
        temps partiel
      </strong>
       
      sont à considérer comme ceux effectués en temps plein.
    </div>
  </div>
  <p
    class="c3"
  >
    De ce fait, avez-vous une modification à faire dans la déclaration du salaire ?
    <span
      aria-label="champs obligatoire"
      class="c4"
    >
      *
    </span>
  </p>
  <div
    class="c5"
  >
    <label
      class="c6"
    >
      <input
        checked=""
        name="hasBrancheNewSalaire"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c6"
    >
      <input
        name="hasBrancheNewSalaire"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
  <p
    class="c3"
  >
    Le salaire mensuel a-t-il été le même durant les 12 derniers mois ?
    <span
      aria-label="champs obligatoire"
      class="c4"
    >
      *
    </span>
  </p>
  <div
    class="c5"
  >
    <label
      class="c6"
    >
      <input
        name="hasBrancheNewRegularSalaire"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c6"
    >
      <input
        checked=""
        name="hasBrancheNewRegularSalaire"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
  <div
    class="c7"
  >
    <div
      class="c8"
    >
      <table
        class="c9 c10"
      >
        <thead>
          <tr>
            <th>
              Mois
            </th>
            <th>
              Salaire mensuel ajusté (primes et temps partiel inclus)
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <label
                for="salaire0"
              >
                juillet 2019
              </label>
            </td>
            <td>
              <div
                class="c11"
              >
                <input
                  class="c12"
                  id="salaire0"
                  name="brancheNewIrregularSalaire[0].salary"
                  type="number"
                  value=""
                />
                <span
                  aria-hidden="true"
                  class="c13"
                >
                  €
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <td>
              <label
                for="salaire1"
              >
                juin 2019
              </label>
            </td>
            <td>
              <div
                class="c11"
              >
                <input
                  class="c12"
                  id="salaire1"
                  name="brancheNewIrregularSalaire[1].salary"
                  type="number"
                  value=""
                />
                <span
                  aria-hidden="true"
                  class="c13"
                >
                  €
                </span>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`<Salaire /> renders an input field of type number 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: auto;
  min-height: 48px;
  color: #3e486e;
  background-color: #fff;
  border-color: #7994d4;
  border-style: solid;
  border-width: 2px;
  border-radius: 0.6rem;
  box-shadow: none;
  -webkit-animation: none;
  animation: none;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 1rem 1.6rem;
  text-align: left;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2em;
  padding: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 2rem;
}

.c3 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: default;
}

.c4 {
  display: inline-block;
  margin-left: 1rem;
  color: #eb5757;
}

.c1 {
  margin-top: 2rem;
}

.c8 {
  width: 10em;
  text-align: right;
  border-color: #bbcadf;
  padding-right: 1.6rem;
  text-align: right;
}

.c8::-webkit-outer-spin-button,
.c8::-webkit-inner-spin-button {
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c7 {
  position: relative;
  display: inline-block;
  margin-right: 2rem;
}

.c9 {
  position: absolute;
  top: 50%;
  right: 0.25rem;
  color: #4d73b8;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

@media (max-width:600px) {
  .c3 {
    font-size: 1.6rem;
  }
}

<div>
  <p>
    Le salaire à renseigner pour cette convention collective inclut:
  </p>
  <ul>
    <li>
      les primes prévues au contrat de travail
    </li>
  </ul>
  <strong>
    mais elle exclut
  </strong>
  :
  <ul>
    <li>
      les heures supplémentaires au delà de l’heure normale de l’entreprise
    </li>
    <li>
      les majorations/indemnités liées à un déplacement ou un détachement
    </li>
  </ul>
  <div
    class="c0 c1"
  >
    <div
      class="c2"
      role="alert"
    >
      Attention, les éventuelles 
      <strong>
        primes
      </strong>
       renseignées précédemment ne seront pas prises en compte dans le calcul de l’indemnité de la convention collective. Si des primes ont été perçues, merci de les intégrer à la rémunération ci-après.
      <br />
      Aussi, si il y en a, les mois effectués en 
      <strong>
        temps partiel
      </strong>
       
      sont à considérer comme ceux effectués en temps plein.
    </div>
  </div>
  <p
    class="c3"
  >
    De ce fait, avez-vous une modification à faire dans la déclaration du salaire ?
    <span
      aria-label="champs obligatoire"
      class="c4"
    >
      *
    </span>
  </p>
  <div
    class="c5"
  >
    <label
      class="c6"
    >
      <input
        checked=""
        name="hasBrancheNewSalaire"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c6"
    >
      <input
        name="hasBrancheNewSalaire"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
  <p
    class="c3"
  >
    Le salaire mensuel a-t-il été le même durant les 12 derniers mois ?
    <span
      aria-label="champs obligatoire"
      class="c4"
    >
      *
    </span>
  </p>
  <div
    class="c5"
  >
    <label
      class="c6"
    >
      <input
        checked=""
        name="hasBrancheNewRegularSalaire"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c6"
    >
      <input
        name="hasBrancheNewRegularSalaire"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
  <p>
    Salaire mensuel ajusté (primes et temps partiel inclus)
  </p>
  <div
    class="c7"
  >
    <input
      class="c8"
      name="brancheNewRegularSalaire"
      type="number"
      value=""
    />
    <span
      aria-hidden="true"
      class="c9"
    >
      €
    </span>
  </div>
</div>
`;
